cql3/statements/use_statement.cc

```diff
@@ -56,7 +56,11 @@ future<> use_statement::check_access(query_processor& qp, const service::client_
 
 future<::shared_ptr<cql_transport::messages::result_message>>
 use_statement::execute(query_processor& qp, service::query_state& state, const query_options& options, std::optional<service::group0_guard> guard) const {
-    state.get_client_state().set_keyspace(qp.db().real_database(), _keyspace);
+    try {
+        state.get_client_state().set_keyspace(qp.db().real_database(), _keyspace);
+    } catch(...) {
+        return make_exception_future<::shared_ptr<cql_transport::messages::result_message>>(std::current_exception());
```

Comment:
Hi, I have 2 FYI's regarding this change that I just saw:
1. You could have used `current_exception_as_future<::shared_ptr<cql_transport::messages::result_message>>()`
2. You could turn this function into a coroutine.

And a potential problem:
3. can't `::make_shared` below throw too?  (2. above would fix hande that and return an exception future in all cases)

It might be fine to throw on "environmental" failures and return a failed future on "business logic" errors, but it's harder to handle errors this way.